library/ziti.c

```diff
@@ -643,23 +643,35 @@ static void ziti_re_auth(ziti_context ztx) {
 
     ziti_re_auth_with_cb(ztx, session_cb, init_req);
 }
-static void set_posture_query_defaults(ziti_service *service){
+
+static void set_posture_query_defaults(ziti_service *service) {
     int posture_set_idx;
-    for(posture_set_idx = 0; service->posture_query_set[posture_set_idx] != 0; posture_set_idx++) {
+    for (posture_set_idx = 0; service->posture_query_set[posture_set_idx] != 0; posture_set_idx++) {
         int posture_query_idx;
-        for(posture_query_idx = 0; service->posture_query_set[posture_set_idx]->posture_queries[posture_query_idx]; posture_query_idx++){
+        for (posture_query_idx = 0; service->posture_query_set[posture_set_idx]->posture_queries[posture_query_idx]; posture_query_idx++) {
 
             //if the controller doesn't support
-            if(service->posture_query_set[posture_set_idx]->posture_queries[posture_query_idx]->timeoutRemaining == NULL) {
+            if (service->posture_query_set[posture_set_idx]->posture_queries[posture_query_idx]->timeoutRemaining == NULL) {
                 //free done by model_free
-                int *timeoutRemaining = calloc(1,sizeof(int));
+                int *timeoutRemaining = calloc(1, sizeof(int));
                 *timeoutRemaining = -1;
                 service->posture_query_set[posture_set_idx]->posture_queries[posture_query_idx]->timeoutRemaining = timeoutRemaining;
             }
         }
     }
 }
 
+// is_service_updated returns 0 if the direct service properties
+// and configurations have not been altered. Will return non-0
+// values if they have. This ignores posture query alterations.
+static int is_service_updated(ziti_service *new, ziti_service *old) {
+    if (strcmp(old->updated_at, new->updated_at) != 0) {
+        return 1;
+    }
+
+    return model_map_compare(&old->config, &new->config, get_model_map_meta());
```

Comment:
pretty sure it is the wrong type_meta to pass in here. It should the type_meta of the values in the map

Reply:
Sure is. Fixed.